src/render/OpenGL.cpp

```diff
@@ -1762,34 +1762,35 @@ void CHyprOpenGLImpl::renderSplash(cairo_t* const CAIRO, cairo_surface_t* const
 void CHyprOpenGLImpl::createBGTextureForMonitor(CMonitor* pMonitor) {
     RASSERT(m_RenderData.pMonitor, "Tried to createBGTex without begin()!");
 
-    static auto* const              PNOSPLASH        = &g_pConfigManager->getConfigValuePtr("misc:disable_splash_rendering")->intValue;
-    static auto* const              PDISABLEHYPRCHAN = &g_pConfigManager->getConfigValuePtr("misc:disable_hypr_chan")->intValue;
-    static auto* const              PFORCEHYPRCHAN   = &g_pConfigManager->getConfigValuePtr("misc:force_hypr_chan")->intValue;
-
-    std::random_device              dev;
-    std::mt19937                    engine(dev());
-    std::uniform_int_distribution<> distribution(0, 2);
-    std::uniform_int_distribution<> distribution2(0, 1);
-
-    const bool                      USEANIME = *PFORCEHYPRCHAN || distribution(engine) == 0; // 66% for anime
+    static auto* const PNOSPLASH       = &g_pConfigManager->getConfigValuePtr("misc:disable_splash_rendering")->intValue;
+    static auto* const PFORCEHYPRCHAN  = &g_pConfigManager->getConfigValuePtr("misc:force_hypr_chan")->intValue;
+    static auto* const PFORCEWALLPAPER = &g_pConfigManager->getConfigValuePtr("misc:force_default_wallpaper")->intValue;
 
     // release the last tex if exists
     const auto PTEX = &m_mMonitorBGTextures[pMonitor];
     PTEX->destroyTexture();
 
     PTEX->allocate();
-
     Debug::log(LOG, "Allocated texture for BGTex");
 
     // TODO: use relative paths to the installation
     // or configure the paths at build time
+    std::string texPath    = "/usr/share/hyprland/wall_";
+    std::string prefixes[] = {"", "anime_", "anime2_"};
 
     // get the adequate tex
-    std::string texPath = "/usr/share/hyprland/wall_";
-    if (!*PDISABLEHYPRCHAN)
-        texPath += std::string(USEANIME ? (distribution2(engine) == 0 ? "anime_" : "anime2_") : "");
-
-    // check if wallpapers exist
+    if (*PFORCEWALLPAPER == -1) {
+        std::random_device              dev;
+        std::mt19937                    engine(dev());
+        std::uniform_int_distribution<> distribution(0, 2);
+        std::uniform_int_distribution<> distribution_anime(1, 2);
+
+        if (PFORCEHYPRCHAN)
+            texPath += prefixes[distribution_anime(engine)];
+        else
+            texPath += prefixes[distribution(engine)];
+    } else
+        texPath += prefixes[*PFORCEWALLPAPER];
```

Comment:
didn't notice this, but this needs a `std::clamp`, if someone puts in 27 it will crash

Reply:
k sure I was already thinking about that.

Is there a way to do that when reading the config? I'd prefer to also give the user a warning if something went wrong. From the code I read it looks like you're only validating the types? Would be kinda stupid to change it for such a small feature tho.

Also sorry for the formatting :/ that was kinda retarded

Reply:
nope we have to clamp here